play-json/shared/src/main/scala/JsResult.scala

```diff
@@ -197,8 +197,33 @@ sealed trait JsResult[+A] { self =>
 }
 
 object JsResult {
+  import scala.util.{ Failure, Try, Success }
   import play.api.libs.functional._
 
+  case class Exception(cause: JsError)
+    extends IllegalArgumentException(Json stringify JsError.toJson(cause))
```

Comment:
I'm just not clear on which method is being passed an invalid argument. See https://docs.oracle.com/javase/8/docs/api/java/lang/IllegalArgumentException.html:
> Thrown to indicate that a method has been passed an illegal or inappropriate argument.